### ISSUE TITLE
require_full_tree vs require_tree after upgrading to 2.1.1 from 1.8.7

### ISSUE THREAD

**Issue by ryanberckmans** [bertramdev/asset-pipeline#225]
Upgraded asset-pipeline to 2.1.1 from 1.8.7,  less-asset-pipeline to 2.0.8 from 1.10.0. No other asset-pipeline plugins. Grails 2.3.8.

Besides my issue below, are there any breaking changes I should be aware of?

Before upgrading, `*= require_tree ../lib/font-awesome` brought in all types of files from dir tree, including css, fonts and js files. 

After upgrading, I had to rewrite that as `*= require_tree /font-awesome`, and avoid the `../lib`, to suppress a ScanDirectory RuntimeException. 

With the new `*= require_tree /font-awesome` syntax, I must use `require_full_tree` to achieve the same behavior instead of `require_tree`.

Most of my require_tree directives are in LESS files. I noticed this line from the guide:

> The require_full_tree directive is an additional directive similar to require_tree except that it will include files from all plugins that contain the relative base path. This is helpful when developing an application that is split into multiple plugins.

Does this mean that `require_tree /foo` in a LESS file will bring in only `/foo/**/*.less`, and not bring in css, javascripts, fonts, etc.?

In what version was this change introduced? Do you recommend using require_full_tree? Thanks.

**TL;DR**

After upgrading, why do I need to use require_full_tree?

```
*= require_tree ../lib/font-awesome   --->   *= require_full_tree /font-awesome
```

**Comment by davydotcom** [comment]
require directives only ever bring in files that are of the same content type as the file in which it was in. For example if using require_tree inside of application.js it will only ever (and has only ever) bring in js files. Require tree uses the same resolver only to bring in these files. It does this so a `require_tree .` would not include your entire js structure and become unwieldy. This was always its intended behavior but it was previously possible to confuse it by using tree traversal with `..`. Now that resolvers are better isolated it is doing more of what it was designed to do. Now, that being said, `require_full_tree` is the counterpart to require_tree and allows you to define a path that scans across all resolvers and include every file and sub file in it matching the same content type. This is potentially useful if writing a plugin that is javascript heavy but you wish to be extensible at the javascript or css level .
